parachain/integration-test/src/lib.rs

```diff
@@ -0,0 +1,379 @@
+		Some(value) => value,
+		None => {
+			panic!("Value not found")
+		},
+	};
+
+	Ok(encoded_value)
+}
+
+/// Configure the state machines and spawn the messaging relayer
+async fn create_clients(
+) -> Result<(SubstrateClient<Hyperbridge>, SubstrateClient<Hyperbridge>), anyhow::Error> {
+	let chain_a_config = SubstrateConfig {
+		state_machine: Kusama(2000),
+		hashing: None,
+		consensus_state_id: Some("PARA".to_string()),
+		rpc_ws: "ws://127.0.0.1:9990".to_string(), // url from local-testnet zombienet config
+		max_rpc_payload_size: None,
+		signer: Some("e5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0".to_string()),
+		latest_height: None,
+		max_concurent_queries: None,
+	};
+
+	let chain_b_config = SubstrateConfig {
+		state_machine: Kusama(2001),
+		hashing: None,
+		consensus_state_id: Some("PARA".to_string()),
+		rpc_ws: "ws://127.0.0.1:9991".to_string(),
+		max_rpc_payload_size: None,
+		signer: Some("e5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0".to_string()),
+		latest_height: None,
+		max_concurent_queries: None,
+	};
+
+	// setup state machines
+	let chain_a_sub_client =
+		SubstrateClient::<KeccakSubstrateChain>::new(chain_a_config.clone()).await?;
+	let chain_b_sub_client =
+		SubstrateClient::<KeccakSubstrateChain>::new(chain_b_config.clone()).await?;
+
+	Ok((chain_a_sub_client, chain_b_sub_client))
+}
+
+/// Assertion is on ismp related events, and state changes on source and destination chain
+/// Alice in chain A sends 100_000 tokens to Alice in chain B
+#[tokio::test(flavor = "multi_thread")]
+async fn submit_transfer_function_works() -> Result<(), anyhow::Error> {
+	log_setup()?;
+	let (chain_a_sub_client, chain_b_sub_client) = create_clients().await?;
+
+	log::info!(
+		"🧊integration test for para:{} to para {}: fund transfer",
+		chain_a_sub_client.clone().state_machine_id().state_id,
+		chain_b_sub_client.clone().state_machine_id().state_id
+	);
+
+	// initiate message relaying task
+	let tx_payment = Arc::new(
+		TransactionPayment::initialize(&"/tmp/dev.db") // out of hyperbridge directory
+			.await
+			.map_err(|err| anyhow!("Error initializing database: {err:?}"))?,
+	);
+
+	let tokio_handle = tokio::runtime::Handle::current();
+	let task_manager = TaskManager::new(tokio_handle, None)?;
+	let relayer_config = RelayerConfig::default();
+
+	let (chain_a_client, chain_b_client) = (
+		Arc::new(chain_a_sub_client.clone()) as Arc<dyn IsmpProvider>,
+		Arc::new(chain_b_sub_client.clone()) as Arc<dyn IsmpProvider>,
+	);
+
+	let mut client_map = HashMap::new();
+	client_map
+		.insert(chain_a_sub_client.clone().state_machine_id().state_id, chain_a_client.clone());
+	client_map
+		.insert(chain_b_sub_client.clone().state_machine_id().state_id, chain_b_client.clone());
+
+	let (client_a, client_b) = (chain_a_sub_client.clone().client, chain_b_sub_client.client);
+
+	relay(
+		chain_a_sub_client,
+		chain_b_client.clone(),
+		relayer_config.clone(),
+		Kusama(3000), // random coprocessor id
```

Comment:
nit: don't pollute the namespace with enum variants, prefer to use the fully qualified enum variant
```suggestion
		StateMachine::Kusama(3000), // random coprocessor id
```